src/Nethermind/Nethermind.Consensus/Validators/TxValidator.cs

```diff
@@ -106,54 +149,119 @@ private bool ValidateSignature(Transaction tx, IReleaseSpec spec)
             return !spec.ValidateChainId;
         }
 
-        private static bool Validate4844Fields(Transaction transaction)
+        private static bool Validate4844Fields(Transaction transaction, ref string? error)
         {
             // Execution-payload version verification
             if (!transaction.SupportsBlobs)
             {
-                return transaction.MaxFeePerBlobGas is null &&
-                       transaction.BlobVersionedHashes is null &&
-                       transaction is not { NetworkWrapper: ShardBlobNetworkWrapper };
+                if (transaction.MaxFeePerBlobGas is not null)
+                {
+                    error = TxErrorMessages.NotAllowedMaxFeePerBlobGas;
+                    return false;
+                }
+
+                if (transaction.BlobVersionedHashes is not null)
+                {
+                    error = TxErrorMessages.NotAllowedBlobVersionedHashes;
+                    return false;
+                }
+
+                if (transaction is { NetworkWrapper: ShardBlobNetworkWrapper })
+                {
+                    //This must be an internal issue?
+                    error = TxErrorMessages.InvalidTransaction;
+                    return false;
+                }
+
+                return true;
+            }
+
+            if (transaction.To is null)
+            {
+                error = TxErrorMessages.TxMissingTo;
+                return false;
+            }
+
+            if (transaction.MaxFeePerBlobGas is null)
+            {
+                error = TxErrorMessages.BlobTxMissingMaxFeePerBlobGas;
+                return false;
+            }
+
+            if (transaction.BlobVersionedHashes is null)
+            {
+                error = TxErrorMessages.BlobTxMissingBlobVersionedHashes;
+                return false;
             }
 
-            if (transaction.To is null ||
-                transaction.MaxFeePerBlobGas is null ||
-                transaction.BlobVersionedHashes is null ||
-                BlobGasCalculator.CalculateBlobGas(transaction.BlobVersionedHashes!.Length) > Eip4844Constants.MaxBlobGasPerTransaction ||
-                transaction.BlobVersionedHashes!.Length < Eip4844Constants.MinBlobsPerTransaction)
+            var totalDataGas = BlobGasCalculator.CalculateBlobGas(transaction.BlobVersionedHashes!.Length);
+            if (totalDataGas > Eip4844Constants.MaxBlobGasPerTransaction)
             {
+                error = TxErrorMessages.BlobTxGasLimitExceeded;
+                return false;
+            }
+            if (transaction.BlobVersionedHashes!.Length < Eip4844Constants.MinBlobsPerTransaction)
+            {
+                error = TxErrorMessages.BlobTxMissingBlobs;
                 return false;
             }
 
             int blobCount = transaction.BlobVersionedHashes.Length;
```

Comment:
May be even useful above on L197, L203

Reply:
I am not sure what you mean here?

Reply:
this property is accessed earlier, lets put it in variable earlier